zebra-rpc/src/server.rs

```diff
@@ -26,21 +26,32 @@ pub struct RpcServer;
 
 impl RpcServer {
     /// Start a new RPC server endpoint
-    pub fn spawn<Mempool>(
+    pub fn spawn<Mempool, State>(
         config: Config,
         app_version: String,
         mempool: Buffer<Mempool, mempool::Request>,
+        state: Buffer<State, zebra_state::Request>,
     ) -> tokio::task::JoinHandle<()>
     where
         Mempool: tower::Service<mempool::Request, Response = mempool::Response, Error = BoxError>
             + 'static,
         Mempool::Future: Send,
+        State: Service<
+                zebra_state::Request,
+                Response = zebra_state::Response,
+                Error = zebra_state::BoxError,
+            > + 'static,
+        State::Future: Send,
     {
         if let Some(listen_addr) = config.listen_addr {
             info!("Trying to open RPC endpoint at {}...", listen_addr,);
 
             // Initialize the rpc methods with the zebra version
-            let rpc_impl = RpcImpl::new(app_version, mempool);
+            let rpc_impl = RpcImpl {
+                app_version,
+                mempool,
+                state,
+            };
```

Comment:
Minor: maybe we can still use the `new` method here?

```suggestion
            let rpc_impl = RpcImpl::new(app_version, mempool, state);
```

Reply:
Done in https://github.com/ZcashFoundation/zebra/pull/3707/commits/6a4055f2030fb896b89f9c015edb6aa0c97be606